osu.Game.Rulesets.Mania/Objects/Drawables/DrawableNote.cs

```diff
@@ -34,6 +47,20 @@ public DrawableNote(Note hitObject)
             });
         }
 
+        [BackgroundDependencyLoader(true)]
+        private void load(ManiaRulesetConfigManager rulesetConfig, BeatDivisorFinder beatDivisorFinder)
+        {
+            if (beatDivisorFinder != null)
+            {
+                rulesetConfig?.BindWith(ManiaRulesetSetting.ColourCodedNotes, configColourCodedNotes);
+
+                HitObject.StartTimeBindable.BindValueChanged(_ => snap.Value = beatDivisorFinder.FindDivisor(HitObject), true);
+
+                snap.BindValueChanged(_ => updateSnapColour(), true);
+                configColourCodedNotes.BindValueChanged(_ => updateSnapColour());
```

Comment:
Putting `BindValueChanged` hooks inside a BDL load method is generally frowned upon. Probably safest to move these to `LoadComplete`.

Reply:
Also just for legibility, I'd move the `BindValueChanged` with the `true` specification to be the last one in the method. That's how we do it everywhere else.

Reply:
this one is on me, not sure how I missed the callbacks.